kore/test/Test/Kore/Step/Rule/Simplify.hs

```diff
@@ -293,34 +304,60 @@ test_simplifyClaimRule =
       where
-        run = runNoSMT . runSimplifier env
+        run =
+            runNoSMT
+            . runSimplifier env
+            . flip runReaderT TestEnv { replacements, input, requires }
+            . runTestSimplifierT
         env =
             Mock.env
-                { simplifierTermLike
-                , simplifierCondition = emptyConditionSimplifier
+                { simplifierCondition = emptyConditionSimplifier
                 , simplifierAxioms = mempty
                 }
-        simplifierTermLike = termLikeSimplifier $ \sideCondition termLike -> do
-            let rule = getOnePathRule input
-                left = Lens.view (field @"left") rule
-                sort = termLikeSort left
-                expectSideCondition =
-                    makeAndPredicate requires (makeCeilPredicate sort left)
-                    & liftPredicate
-                    & Predicate.coerceSort predicateSort
-                    & Condition.fromPredicate
-                    & SideCondition.fromCondition
-                satisfied = sideCondition == expectSideCondition
-            return
-                . OrPattern.fromTermLike
-                . (if satisfied then applyReplacements else id)
-                $ termLike
 
+data TestEnv =
+    TestEnv
+    { replacements :: ![(TermLike VariableName, TermLike VariableName)]
+    , input :: !OnePathRule
+    , requires :: !(Predicate VariableName)
+    }
+
+newtype TestSimplifierT m a =
+    TestSimplifierT { runTestSimplifierT :: ReaderT TestEnv m a }
+    deriving (Functor, Applicative, Monad)
+    deriving (MonadReader TestEnv)
+    deriving (MonadLog, MonadSMT)
+
+instance MonadTrans TestSimplifierT where
+    lift = TestSimplifierT . lift
+
+instance MFunctor TestSimplifierT where
+    hoist f = TestSimplifierT . hoist f . runTestSimplifierT
+
+instance MonadSimplify m => MonadSimplify (TestSimplifierT m) where
+    simplifyTermLike sideCondition termLike = do
+        TestEnv { replacements, input, requires } <- Reader.ask
+        let rule = getOnePathRule input
+            left = Lens.view (field @"left") rule
+            sort = termLikeSort left
+            expectSideCondition =
+                makeAndPredicate requires (makeCeilPredicate sort left)
+                & liftPredicate
+                & Predicate.coerceSort predicateSort
+                & Condition.fromPredicate
+                & SideCondition.fromCondition
+            satisfied = sideCondition == expectSideCondition
+        return
+            . OrPattern.fromTermLike
+            . (if satisfied then applyReplacements replacements else id)
+            $ termLike
+      where
         applyReplacements
             :: InternalVariable variable
-            => TermLike variable
+            => [(TermLike VariableName, TermLike VariableName)]
+            -> TermLike variable
             -> TermLike variable
-        applyReplacements zero =
+        applyReplacements replacements zero =
```

Comment:
I know this isn't your change, but what are these replacements?

Reply:
This test uses a mock simplifier: instead of actually running the simplifier, it just replaces certain patterns in the result. This is the list of patterns to replace, and their replacements.